pkg/virt-launcher/virtwrap/network/dhcp/dhcp.go

```diff
@@ -31,7 +34,15 @@ import (
 	"kubevirt.io/kubevirt/pkg/log"
 )
 
-const infiniteLease = 999 * 24 * time.Hour
+const (
+	infiniteLease             = 999 * 24 * time.Hour
+	errorSearchDomainNotValid = "Search domain is not valid"
+	errorSearchDomainTooLong  = "Search domains length exceeded allowable size"
+)
+
+// simple domain validation regex. Put it here to avoid compiling each time.
+// Note this requires that unicode domains be presented in their ASCII format
+var searchDomainValidationRegex = regexp.MustCompile(`^(?:[_a-z0-9](?:[_a-z0-9-]{0,61}[a-z0-9])?\.)+(?:[a-z](?:[a-z0-9-]{0,61}[a-z0-9])?)?$`)
```

Comment:
If I'm reading this regex correctly, It appears that it's enforcing that labels cannot end with a hyphen or underscore. That's definitely correct for hyphens, but I am unsure of underscores. (RFC 1123 specifically supersedes RFC 952 with regards to [digits in] the first character, but as near as I can tell is ambiguous with regards to subsequent ones).